airbyte-cdk/bulk/toolkits/load-db/src/main/kotlin/io/airbyte/cdk/load/orchestration/legacy_typing_deduping/TypingDedupingSqlGenerator.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.orchestration.legacy_typing_deduping
+
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.orchestration.ColumnNameMapping
+import io.airbyte.cdk.load.orchestration.Sql
+import io.airbyte.cdk.load.orchestration.TableName
+import io.airbyte.cdk.load.orchestration.TableNames
+import io.airbyte.cdk.load.orchestration.TableNames.Companion.SOFT_RESET_SUFFIX
+import java.time.Instant
+
```

Comment:
pretty clear analogue to the old SqlGenerator. Interface is updated to use DestinationStream x TableName(s) x ColumnNameMapping, and some methods renamed for clarity.